cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -180,25 +180,18 @@ endif()
 find_package(Thrift)
 
 if (NOT THRIFT_FOUND)
-
   set(THRIFT_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/thrift_ep/src/thrift_ep-install")
   set(THRIFT_HOME "${THRIFT_PREFIX}")
   set(THRIFT_INCLUDE_DIR "${THRIFT_PREFIX}/include")
-  IF (${UPPERCASE_BUILD_TYPE} STREQUAL "DEBUG")
-    IF (MSVC)
-      set(THRIFT_STATIC_LIB "${THRIFT_PREFIX}/lib/thriftmdd.lib")
-    ELSE()
-      set(THRIFT_STATIC_LIB "${THRIFT_PREFIX}/lib/libthriftd.a")
-    ENDIF()
-  ELSE()
-    IF (MSVC)
-      set(THRIFT_STATIC_LIB "${THRIFT_PREFIX}/lib/thriftmd.lib")
-    ELSE()
-      set(THRIFT_STATIC_LIB "${THRIFT_PREFIX}/lib/libthrift.a")
-    ENDIF()
-  ENDIF()
+  set(THRIFT_STATIC_LIB_NAME "${CMAKE_STATIC_LIBRARY_PREFIX}thrift")
+  if (MSVC)
+    set(THRIFT_STATIC_LIB_NAME "${THRIFT_STATIC_LIB_NAME}${THRIFT_MSVC_STATIC_LIB_SUFFIX}")
```

Comment:
`THRIFT_MSVC_STATIC_LIB_SUFFIX` is configurable by users and should be used to found custom thrift builds made by users. Here we build thrift from the sources and have full control of what we want to have at the output. It seems better to use hardcoded `md` suffix here for MSVC. Usage of `THRIFT_MSVC_STATIC_LIB_SUFFIX` might break the build, if user will overwrite the default value, but installed thrift will be not found.